pkg/river/token/builder/builder.go

```diff
@@ -166,9 +171,21 @@ func (b *Body) encodeFields(rv reflect.Value) {
 	}
 
 	fields := rivertags.Get(rv.Type())
+	clone := reflect.New(rv.Type()).Elem()
+
+	var goRiverDefaulter = reflect.TypeOf((*value.Defaulter)(nil)).Elem()
+	if rv.CanAddr() && rv.Addr().Type().Implements(goRiverDefaulter) {
+		clone.Addr().Interface().(value.Defaulter).SetToDefault()
+	}
```

Comment:
Because we're calling SetToDefault on `clone`, we should be checking whether `clone.CanAddr()` is true rather than `rv.CanAddr()`; the two aren't linked, so it's possible that clone could be addressible when rv isn't. 

In fact, clone will _always_ be addressible because we create it as a pointer above with `reflect.New`.

Given that we need to check the addressability of clone instead, it may make more sense to make the whole condition about clone instead of rv:

```suggestion
	if clone.CanAddr() && clone.Addr().Type().Implements(goRiverDefaulter) {
		clone.Addr().Interface().(value.Defaulter).SetToDefault()
	}
```